tests/MongoDB.Driver.Tests/Specifications/unified-test-format/UnifiedErrorMatcher.cs

```diff
@@ -0,0 +1,145 @@
+        }
+
+        private void AssertErrorCodeName(Exception actualException, string expectedErrorCodeName)
+        {
+            var mongoCommandException = actualException.Should().BeOfType<MongoCommandException>().Subject;
+
+            mongoCommandException.CodeName.Should().Be(expectedErrorCodeName);
+        }
+
+        private void AssertErrorContains(Exception actualException, string expectedSubstring)
+        {
+            actualException.Message.Should().ContainEquivalentOf(expectedSubstring);
+        }
+
+        private void AssertErrorLabelsContain(Exception actualException, IEnumerable<string> expectedErrorLabels)
+        {
+            actualException.Should().BeAssignableTo<MongoException>();
+            var mongoException = actualException as MongoException;
+
+            mongoException.ErrorLabels.Should().Contain(expectedErrorLabels);
+        }
+
+        private void AssertErrorLabelsOmit(Exception actualException, IEnumerable<string> expectedAbsentErrorLabels)
+        {
+            actualException.Should().BeAssignableTo<MongoException>();
+            var mongoException = actualException as MongoException;
+
+            mongoException.ErrorLabels.Should().NotContain(expectedAbsentErrorLabels);
+        }
+
+        private void AssertExpectResult(Exception actualException, BsonDocument expectedResult)
+        {
+            actualException.Should().BeOfType<MongoBulkWriteException<BsonDocument>>();
+            var bulkWriteException = actualException as MongoBulkWriteException<BsonDocument>;
+            var bulkWriteResult = bulkWriteException.Result;
+            var actualResult = new BsonDocument
+            {
+                { "deletedCount", bulkWriteResult.DeletedCount },
+                { "insertedCount", bulkWriteResult.InsertedCount },
+                { "matchedCount", bulkWriteResult.MatchedCount },
+                { "modifiedCount", bulkWriteResult.ModifiedCount },
+                { "upsertedCount", bulkWriteResult.Upserts.Count },
+                { "upsertedIds", new BsonDocument(bulkWriteResult.Upserts.Select(x => new BsonElement(x.Index.ToString(), x.Id))) }
+            };
+
+            actualResult.Should().Be(expectedResult);
+        }
+
+        private void AssertIsClientError(Exception actualException, bool expectedIsClientError)
+        {
+            var actualIsClientError =
+                actualException is MongoClientException ||
+                actualException is BsonException;
+
+            if (actualIsClientError != expectedIsClientError)
+            {
+                var message = $"Expected exception to {(expectedIsClientError ? "" : "not ")}be a client exception, but found {actualException}.";
+
+                throw new AssertionException(message);
+            }
+        }
+
+        private void AssertIsError(Exception actualException, bool expectedIsError)
+        {
+            if (!expectedIsError)
```

Comment:
Consider:

            if (expectedIsError == false)

p.s. I know that using `!` is more common, **but** in some cases (like here) it reads better to compare to false.